test/integration/test_connection_management.py

```diff
@@ -105,3 +105,23 @@ def test_http_h2_connection_management_single_request_per_conn_1(http_test_serve
 @pytest.mark.skipif(isSanitizerRun(), reason="Unstable in sanitizer runs")
 def test_http_h2_connection_management_single_request_per_conn_1(http_test_server_fixture):
   connection_management_test_request_per_connection(http_test_server_fixture, 5, True)
+
+
+def test_h1_pool_strategy(http_test_server_fixture):
+  """
+  Test that with the "HOT" strategy only the first created connection gets to send requests.
+  Then, with the "FAIR" strategy, we expect the other connection to be used as well.
```

Comment:
If we're adverting fairness, I think we want to show that there is roughly equal number of C0/C1 across a run of a fair few RPS. Do you think that could be tested easily?

Reply:
Actually, yes, so done in https://github.com/envoyproxy/nighthawk/pull/198/commits/cf86b00ae5648d81afad563b975611999a0ee602